### PR TITLE
Image De-Photobombing

### PR DESCRIPTION
**Pull request by hiteshhhh007**
# Pull Request for DL-Simplified 💡

## Issue Title : Image De-Photobombing

- **Info about the related issue (Aim of the project)** : 
- **Name:** Hitesh
- **GitHub ID:** hiteshhhh007
- **Email ID:** hiteshkrishna43@gmail.com
- **Idenitfy yourself:** `GSSoC'24 Participant`

Closes: #518 will be closed through this PR

### Describe the add-ons or changes you've made 📃

I have added 3 different approaches and all the 3 approaches have been tested out and then was decided to be implemented for this project. The 3 approaches are:
1. OpenCV-Technique
2. Usage of U-Net DL Architecture
3. Pre-Trained Architecture DeepFillv2 using Gated NN's.
All the approaches are implemented in the .ipynb with comments and details.

## Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

All the approaches have been tested out on multiple images on metrics like loss and accuracy. The loss metric has been more focused here because of the project nature.

Models have been verified by their prediction on the val-set and then on test-set, both of which was taken from the training-set

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — APPROVED** [review]
Approved @hiteshhhh007

**Comment by hiteshhhh007** [comment]
Please Merge the PR @abhisheks008
